### PR TITLE
[#294, #295] Update Gmail policies 5 and 7 to include all instances of spoofing and authentication settings that are not complaint

### LINKED ISSUE #294: Full Report Details for GWS.GMAIL.5.5
# 🐛 Summary #
Currently (as of https://github.com/cisagov/ScubaGoggles/pull/278), the report details for GWS.GMAIL.5.5 doesn't always show the full picture.

For example, after setting the action for encrypted attachments from untrusted senders, attachments with scripts from untrusted senders, and anomalous attachment types all to warning, the report just says:

> [OU name]: Emails with encrypted attachments from untrusted senders are kept in the inbox.

Ideally it would report that all three settings are non-compliant, not just one of them. For example:
> [OU name]: The action for encrypted attachments from untrusted senders is set to warning. The action for attachments with scripts from untrusted senders is set to warning. The action for anomalous attachment type is set to warning.

## To reproduce ##
1. Admin center -> Apps -> Google Workspace -> Gmail -> Safety -> Attachments -> Set the action for multiple categories to "Keep in inbox and show warning"
2. Run ScubaGoggles. Note that only one category is reported in the report details column.

## Expected behavior ##
All categories would be listed in the Report Details column.

## Any helpful log output or screenshots ##
Refer to GWS.CHAT.7.1 for an example of how to implement Report Details messages like this in Rego.

### LINKED ISSUE #295: Full Report Details for GWS.GMAIL.7.6
# 🐛 Summary #
Currently (as of https://github.com/cisagov/ScubaGoggles/pull/278), the report details for GWS.GMAIL.7.6 doesn't always show the full picture.

## To reproduce ##
See https://github.com/cisagov/ScubaGoggles/issues/294 for discussion on a related applicable issue.

### LINKED ISSUE #396: GWS.GMAIL.5.5 has the wrong criticality in the code
# 🐛 Summary #
GWS.GMAIL.5.5 is a "SHALL" control, but the criticality listed in the Rego is "Should."

## To reproduce ##
1. Change one of the attachment settings to "keep in inbox"
2. Run ScubaGoggles. Note that 5.5 shows as a "Warning" not a "Failure"

## Expected behavior ##
5.5 non-compliance to be reported as a failure.

## Any helpful log output or screenshots ##
image

### LINKED ISSUE #397: GWS.GMAIL.7.6 has the wrong criticality in the code
# 🐛 Summary #
GWS.GMAIL.7.6 is a "SHALL" control, but the criticality listed in the Rego is "Should."

## To reproduce ##
1. Change one of the spoofing settings to "keep in inbox"
2. Run ScubaGoggles. Note that 7.6 shows as a "Warning" not a "Failure"

## Expected behavior ##
5.5 non-compliance to be reported as a failure.

## Any helpful log output or screenshots ##
image

### PR DESCRIPTION
**Pull request by snarve**
## 🗣 Description ##

Update Gmail policy 5 and 7 to include all settings related to spoofing and authentication that are not complaint

Closes #294 
Closes #295
Closes #396
Closes #397

### 💭 Motivation and context

- The current implementation reports the first failing setting for spoofing and authentication from policies 5 and 7
- So in cases where multiple settings fail only one of them is reported instead of all
- This update handles this use case so that all the failing settings are reported for both the policies

## 🧪 Testing

- Test Gmail policies 5 and 9 (follow the baseline implementation steps)
- Modify the settings for both the policies 
- Make more than one setting non-complaint, ie: leave it at the default value, so more than one setting is set to 'Keep in inbox'
- The report should include all the settings that fail
- Test the unit tests to confirm they pass

Policy 5 can be set as below:

image

Policy 7 can be set as below:

image

Report

image

image

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] If applicable, *All* future TODOs are captured in issues, which are referenced in the PR description.
- [x] The relevant issues PR resolves are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels have been added.
- [x] I have read and agree to the [CONTRIBUTING.md](https://github.com/cisagov/ScubaGoggles/blob/main/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge Checklist

- [ ] This PR has been smoke tested to ensure main is in a functional state when this PR is merged.
- [ ] Squash all commits into one PR level commit using the `Squash and merge` button.

## ✅ Post-merge Checklist

- [ ] Delete the branch to clean up.
- [ ] Close issues resolved by this PR if the [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) did not activate.

### CODE REVIEW

**Review by adhilto — CHANGES_REQUESTED** [review]
Huge improvement over what we had before. Just a few things:

File: rego/Gmail.rego [hunk 1]
```diff
@@ -1209,63 +1253,98 @@ NoSuchEvent7_6 := true if {
...
+EmployeeNamesMessage(NewValueEmployeeNames) := "Inbound emails spoofing employee names"
+    if { NewValueEmployeeNames == "Show warning" }
+    else := ""
+
+InboundEmailsMessage(NewValueInboundEmail) := "Inbound spoofing emails"
+    if { NewValueInboundEmail == "Show warning" }
+    else := ""
+
+UnauthenticatedEmailsMessage(NewValueUnauthenticatedEmails) := "Unauthenticated emails"
+    if { NewValueUnauthenticatedEmails in ["Show warning", "No action"] == true }
```
[the remark this answers is not in the document]
Comment (snarve) [reply rego/Gmail.rego]: updated

File: rego/Gmail.rego [hunk 2]
```diff
@@ -1209,63 +1253,98 @@ NoSuchEvent7_6 := true if {
...
+
+GroupEmailsMessage(NewValueGroupEmails) := "Inbound spoofing emails addresed to groups"
+    if { NewValueGroupEmails == "Show warning" }
+    else := ""
+
+AllDetailedMessages(NewValueDomainNames, NewValueEmployeeNames, NewValueInboundEmails,
+    NewValueUnauthenticatedEmails, NewValueGroupEmails) := [
+        DomainNamesMessage(NewValueDomainNames), EmployeeNamesMessage(NewValueEmployeeNames),
+        InboundEmailsMessage(NewValueInboundEmails), UnauthenticatedEmailsMessage(NewValueUnauthenticatedEmails),
+        UnauthenticatedEmailsMessage(NewValueUnauthenticatedEmails), GroupEmailsMessage(NewValueGroupEmails)
```
[the remark this answers is not in the document]
Comment (snarve) [reply rego/Gmail.rego]: updated

File: rego/Gmail.rego [hunk 3]
```diff
@@ -1209,63 +1253,98 @@ NoSuchEvent7_6 := true if {
...
-    else := "Emails flagged by the spoofing and authentication controls are not kept in the inbox"
+DomainNamesMessage(NewValueDomainNames) := "Inbound emails spoofing domain names"
+    if { NewValueDomainNames == "Show warning" }
+    else := ""
+
+EmployeeNamesMessage(NewValueEmployeeNames) := "Inbound emails spoofing employee names"
+    if { NewValueEmployeeNames == "Show warning" }
+    else := ""
+
+InboundEmailsMessage(NewValueInboundEmail) := "Inbound spoofing emails"
```
[the remark this answers is not in the document]
Comment (adhilto) [reply rego/Gmail.rego:1264]: Maybe call this one "emails spoofing your domain name" and the other one "emails spoofing similar domain names" to clarify the difference between the two domain name ones?

[the remark this answers is not in the document]
Comment (snarve) [reply rego/Gmail.rego]: done

**Review by adhilto — APPROVED** [review]
Nice work

**Review by LaurenBassett — APPROVED** [review]
Much better. Thank you!

**Pull request merged**